delivery/repositories/project_repository.py

```diff
@@ -49,3 +53,158 @@ def get_project(self, project_name):
+                    sample_file.sample_path,
+                    project.path)] if sample_file.checksum else None
+
+        def _sample_checksums(sample):
+            for sample_file in sample.sample_files:
+                yield _sample_file_checksum(sample_file)
+
+        checksum_path = os.path.join(project.path, "checksums.md5")
+        self.filesystem_service.write_checksum_file(
+            checksum_path,
+            {path: checksum for sample in project.samples for checksum, path in _sample_checksums(sample) if checksum})
+
+        return checksum_path
+
+    def get_projects(self, runfolder):
+        """
+        Returns a list of RunfolderProject instances, representing all projects found in this runfolder.
+
+        :param runfolder: a Runfolder instance
+        :return: a list of RunfolderProject instances or None if no projects were found
+        :raises: ProjectsDirNotfoundException if the Unaligned directory could not be found in the runfolder
+        """
+        def dir_contains_fastq_files(d):
+            return any(
+                map(
+                    lambda f: f.endswith("fastq.gz"),
+                    self.filesystem_service.list_files_recursively(d)))
+
+        def project_from_dir(d):
+            project = RunfolderProject(
+                name=os.path.basename(d),
+                path=os.path.join(projects_base_dir, d),
+                runfolder_path=runfolder.path,
+                runfolder_name=runfolder.name
+            )
+            project.samples = self.sample_repository.get_samples(project, runfolder)
+            return project
+
+        try:
+            projects_base_dir = os.path.join(runfolder.path, self.PROJECTS_DIR)
+
+            # only include directories that have fastq.gz files beneath them
+            project_directories = filter(
+                dir_contains_fastq_files,
+                self.filesystem_service.find_project_directories(projects_base_dir)
+            )
+
+            return list(map(project_from_dir, project_directories)) or None
+
+        except FileNotFoundError:
+            raise ProjectsDirNotfoundException("Did not find Unaligned folder for: {}".format(runfolder.name))
+
+    def get_report_files(self, project):
+        """
+        Gets the paths to files associated with the supplied project's report. This can be either a MultiQC report or,
+        if no such report was found, a Sisyphus report.
+
+        :param project: a RunfolderProject instance
+        :return: a tuple with the path to the directory containing the report files and a list of the paths to the
+        report files
+        :raises ProjectReportNotFoundException: if no MultiQC or Sisyphus report was found for the project
+        """
+        if self.filesystem_service.exists(self.multiqc_report_path(project)):
+            return self.multiqc_report_files(project)
+        for sisyphus_report_path in self.sisyphus_report_path(project):
```

Comment:
I don't really get here why there is a single path for multiqc but multiple once for Sisyhus. I guess there is good reason for it though...

Reply:
the reason is that I wanted the method to make sense regardless if the `project` object passed represents an organized project or an unorganized project. For MultiQC-reports, the path (relative to `project.path`) will be the same for both cases but for sisyphus reports, there will be different paths.  A more explicit way could of course be to subclass `project` with separate classes for organized and unorganized `projects` but I think it's nicer if that logic is in the `project_repository`